ui/pages/editor.tsx

```diff
@@ -1,110 +1,254 @@
 import React, { useEffect, useRef, useState } from "react";
-import Editor from "@monaco-editor/react";
+import Editor, { monaco } from "@monaco-editor/react";
 import { FileTree, IFileTreeNode, FileNode, DirectoryNode } from "../components/react/file-tree";
 import path from "path";
-import useSWR from 'swr';
+import useSWR, { mutate } from 'swr';
+import { Icon, Intent } from "@blueprintjs/core";
+import { IconNames } from "@blueprintjs/icons";
+import { v4 as uuidv4 } from 'uuid';// TODO: Import monaco types
+
+interface IEditorModelOptions {
+  onIsDirtyChange?:(editor:EditorModel) => void;
+  onModelSave?:() => void;
+}
 
-export default function EditorPage(props) {
+class EditorModel {
```

Comment:
This is basically a wrapper for the monaco model returned from CreateModel() - helpful to wrap the dirty watch logic in here